app/admin/allocation_chart.rb

```diff
@@ -11,9 +13,36 @@
           column(I18n.t('name'), :user)
           column(I18n.t('skills')) { |allocation| allocation.user.skills.map(&:title).join(', ') }
           column(I18n.t('client'), :project)
-          column(I18n.t('allocated_until'), :end_at)
+          column(I18n.t('allocated_until'), class: 'allocated-column') do |allocation|
+            build_allocation_status_cell(allocation)
+          end
         end
       end
     end
   end
 end
+
+def build_allocation_status_cell(last_allocation)
+  allocation_status = ActiveAdmin::AllocationChartHelper.allocation_status_for(last_allocation: last_allocation)
+
+  column_cell_base(
+    content: allocation_status_content(last_allocation: last_allocation,
+                                       allocation_status: allocation_status), allocation_status: allocation_status
+  )
+end
+
+def allocation_status_content(last_allocation:, allocation_status:)
+  return last_allocation.end_at if last_allocation.end_at
```

Comment:
I feel like this is something that might be useful as a method on the `Allocation` model. That way, the method name will describe what we want when checking if the `end_at` field exists and will make the code more readable. So how about we add this in `app/models/allocation.rb`?

```ruby
def currently_allocated?
  last_allocation.end_at?
end
```

Then the code in here would be:
```ruby
return last_allocation.end_at if last_allocation.currently_allocated?
```

Reply:
An allocation does not need and `end_at` date. So I belive that this method can lead to confusion. What about this method simplification?

```rb
def allocation_status_content(last_allocation:, allocation_status:)
      last_allocation.end_at || unallocated?(allocation_status) ? I18n.t('not_allocated') : I18n.t('allocated')
end
```

Reply:
> An allocation does not necessarily needs an end_at date

Ok, the method name was just an idea, what is it that we're checking for exactly then? when does an allocation have an end_date?

Reply:
Deciding if the column content is `end_at` date when it exists or a text based on allocation status. `end_date` is setting in allocation creation/update.

Reply:
Ok, so I'd say we leave as is